aya/src/maps/mod.rs

```diff
@@ -166,6 +170,24 @@ impl Map {
             name: self.obj.name.clone(),
         })
     }
+
+    pub fn pin<P: AsRef<Path>>(&self, path: P) -> Result<(), MapError> {
+        let fd = self.fd_or_err()?;
+        let path_string = match CString::new(path.as_ref().to_str().unwrap()) {
```

Comment:
I think it would be better to use `to_string_lossy` here than unwrap. Also use `map_err` please instead of matching ok and err.